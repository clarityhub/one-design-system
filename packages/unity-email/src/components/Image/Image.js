import React from 'react';
import { MjmlImage } from 'mjml-react';

const Footer = (props) => (
	<MjmlImage {...props} />
);

export default Footer;
