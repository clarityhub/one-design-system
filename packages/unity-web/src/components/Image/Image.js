import React from 'react';
import { bool, object, string } from 'prop-types';
import { css } from '@emotion/core';
import styled from '@emotion/styled';
import { Img } from 'react-image';

const ImageWrapper = styled.div`
    img {
        box-shadow: 0 2px 4px 0 rgba(0,0,0,.1);
        max-width: 100%;
    }

    ${({ withSpacing }) => withSpacing && css`
        padding: 1rem;
    `}

    ${({ center }) => center && css`
        margin-left: auto;
        margin-right: auto;
        text-align: center;
    `}
`;

const Image = ({ alt, withSpacing, center, wrapperProps = {}, src, loaderUrl, fallbackUrl, ...rest }) => (
	<ImageWrapper {...wrapperProps} withSpacing={withSpacing} center={center}>
		<Img
			src={[
				src,
				fallbackUrl,
			].filter(Boolean)}
			loader={loaderUrl && <img alt={alt} src={loaderUrl} />}
			unloader={fallbackUrl && <img alt={alt} src={fallbackUrl} />}
			alt={alt}
			{...rest}
		/>
	</ImageWrapper>
);

/**
 * Hack for docz and react-docgen
 */
const ImagePropTypes = () => <div />;
ImagePropTypes.propTypes = {
	alt: string.isRequired,
	center: bool,
	fallbackUrl: string,
	loaderUrl: string,
	src: string.isRequired,
	withSpacing: bool,
	wrapperProps: object,
};
Image.propTypes = ImagePropTypes.propTypes;
export { ImagePropTypes };

export default Image;
