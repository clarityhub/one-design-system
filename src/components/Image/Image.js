import React from 'react';
import { bool, object, string } from 'prop-types';
import { css } from '@emotion/core';
import styled from '@emotion/styled';

const ImageWrapper = styled.div`
    img {
        box-shadow: 0 2px 4px 0 rgba(0,0,0,.1);
        max-width: 100%;
    }

    ${({ withSpacing }) => withSpacing && css`
        padding: 1rem;
    `}

    ${({ center }) => center && css`
        margin-left: auto;
        margin-right: auto;
        text-align: center;
    `}
`;

const Image = ({ alt, withSpacing, center, wrapperProps = {}, ...rest }) => (
	<ImageWrapper {...wrapperProps} withSpacing={withSpacing} center={center}>
		<img alt={alt} {...rest} />
	</ImageWrapper>
);

Image.propTypes = {
	alt: string.isRequired,
	center: bool,
	src: string.isRequired,
	withSpacing: bool,
	wrapperProps: object,
};

export default Image;
