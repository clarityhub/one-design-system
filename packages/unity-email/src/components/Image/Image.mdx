---
menu: Components
name: Image
---

import { Playground, PropsTable } from 'docz';
import Image from './Image';

# Email Image

```js
import Image from '@clarityhub/unity-email/lib/components/Image';
```

## Prop Types

<PropsTable of={Image} />
