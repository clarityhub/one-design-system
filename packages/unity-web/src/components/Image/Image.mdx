---
name: Image
menu: Components
---

import { Playground, PropsTable } from 'docz';
import Image from './Image';

# Image

## \<Image /\> component

```js
import Image from '@clarityhub/unity-core/components/Image';
```

## Prop Types

<PropsTable of={Image} />

## Examples

<Playground>
    <Image style={{ width: '20rem' }} src="/public/get-started.png" alt="Get Started" />
    <Image style={{ width: '20rem' }} withSpacing src="/public/get-started.png" alt="Get Started" />
    <Image style={{ width: '20rem' }} center src="/public/get-started.png" alt="Get Started" />
    <Image style={{ width: '20rem' }} center withSpacing src="/public/get-started.png" alt="Get Started" />
</Playground>