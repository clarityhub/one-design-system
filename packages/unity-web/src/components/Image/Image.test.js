import React, { Fragment } from 'react';
import renderer from 'react-test-renderer';

import Image from './Image';

describe('<Image />', () => {
	it('renders images', () => {
		const node = (
			<Fragment>
				<Image src="/public/get-started.png" alt="Get Started" />
				<Image withSpacing src="/public/get-started.png" alt="Get Started" />
				<Image center src="/public/get-started.png" alt="Get Started" />
				<Image center withSpacing src="/public/get-started.png" alt="Get Started" />
			</Fragment>
		);
		const tree = renderer.create(node).toJSON();

		expect(tree).toMatchSnapshot();
	});
});