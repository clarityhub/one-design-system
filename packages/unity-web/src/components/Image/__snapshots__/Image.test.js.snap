// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> renders images 1`] = `
Array [
  <div
    className="css-119gdmf-ImageWrapper e1h9019k0"
  />,
  <div
    className="css-12kn8xa-ImageWrapper-ImageWrapper e1h9019k0"
  />,
  <div
    className="css-11rfl6x-ImageWrapper-ImageWrapper e1h9019k0"
  />,
  <div
    className="css-hdn7cq-ImageWrapper-ImageWrapper-ImageWrapper e1h9019k0"
  />,
]
`;
